x-pack/legacy/plugins/snapshot_restore/public/app/sections/home/recovery_list/recovery_table/shards_table.tsx

```diff
@@ -0,0 +1,252 @@
+        totalTimeInMillis ? (
+          <FormattedMessage
+            id="xpack.snapshotRestore.recoveryList.shardTable.durationValue"
+            defaultMessage="{seconds} {seconds, plural, one {second} other {seconds}}"
+            values={{ seconds: Math.ceil(totalTimeInMillis / 1000) }}
+          />
+        ) : (
+          <EuiLoadingSpinner size="m" />
+        ),
+    },
+    {
+      field: 'repository',
+      name: i18n.translate('xpack.snapshotRestore.recoveryList.shardTable.repositoryColumnTitle', {
+        defaultMessage: 'Repository',
+      }),
+    },
+    {
+      field: 'snapshot',
+      name: i18n.translate('xpack.snapshotRestore.recoveryList.shardTable.snapshotColumnTitle', {
+        defaultMessage: 'Snapshot',
+      }),
+    },
+    {
+      field: 'version',
+      name: i18n.translate('xpack.snapshotRestore.recoveryList.shardTable.versionColumnTitle', {
+        defaultMessage: 'Version',
+      }),
+    },
+    {
+      field: 'targetHost',
+      name: i18n.translate('xpack.snapshotRestore.recoveryList.shardTable.targetHostColumnTitle', {
+        defaultMessage: 'Target host',
+      }),
+    },
+    {
+      field: 'targetNode',
+      name: i18n.translate('xpack.snapshotRestore.recoveryList.shardTable.targetNodeColumnTitle', {
+        defaultMessage: 'Target node',
+      }),
+    },
+    {
+      field: 'bytesTotal',
+      name: i18n.translate('xpack.snapshotRestore.recoveryList.shardTable.bytesColumnTitle', {
+        defaultMessage: 'Bytes',
+      }),
+      render: (
+        bytesTotal: SnapshotRecoveryShard['bytesTotal'],
+        { bytesRecovered, bytesPercent }: SnapshotRecoveryShard
+      ) => {
+        return (
+          <EuiToolTip
+            position="top"
+            content={i18n.translate(
+              'xpack.snapshotRestore.recoveryList.shardTable.bytesTooltipLabel',
+              {
+                defaultMessage: 'Recovered: {recovered}, Total: {total}',
+                values: {
+                  recovered: bytesRecovered,
+                  total: bytesTotal,
+                },
+              }
+            )}
+          >
+            <EuiText size="xs" textAlign="center" style={{ width: '100%' }}>
+              <EuiProgress
+                value={bytesTotal === 0 ? 1 : bytesRecovered}
+                max={bytesTotal === 0 ? 1 : bytesTotal}
+                size="xs"
+              />
+              <EuiSpacer size="xs" />
+              {bytesPercent}
+            </EuiText>
+          </EuiToolTip>
+        );
+      },
+    },
+    {
+      field: 'filesTotal',
+      name: i18n.translate('xpack.snapshotRestore.recoveryList.shardTable.filesColumnTitle', {
+        defaultMessage: 'Files',
+      }),
+      render: (
+        filesTotal: SnapshotRecoveryShard['filesTotal'],
+        { filesRecovered, filesPercent }: SnapshotRecoveryShard
+      ) => {
+        return (
+          <EuiToolTip
```

Comment:
Since this is more than a couple lines and duplicated 3x, how about extracting it into a component?

```jsx
const Progress = ({ total, recovered, percent }) => (
  <EuiToolTip
    position="top"
    content={i18n.translate(
      'xpack.snapshotRestore.recoveryList.shardTable.filesTooltipLabel',
      {
        defaultMessage: 'Recovered: {recovered}, Total: {total}',
        values: {
          recovered,
          total,
        },
      }
    )}
  >
    <EuiText size="xs" textAlign="center" style={{ width: '100%' }}>
      <EuiProgress
        value={total === 0 ? 1 : recovered}
        max={total === 0 ? 1 : total}
        size="xs"
      />
      <EuiSpacer size="xs" />
      {percent}
    </EuiText>
  </EuiToolTip>
);
```

Consuming it will look like this:

```jsx
<Progress total={filesTotal} recovered={filesRecovered} percent={filesPercent} />
```